### DISCUSSION TITLE
Berry for a beginner...

### DISCUSSION THREAD

**Discussion by trlafleur** [berry-lang/berry#231] [Q&A]
I'm very new to Barry... Trying to learn the basics...  

When I copy this test code into the Berry console on the ESP32, the first parts work and get my sensor data printed, the 2nd part is not added the data to the MQTT packet for the next TelePeriod..

or is this the correct way to send/add new MQTT data?

~~~
import json
import string
sensors=json.load(tasmota.read_sensors())
print(sensors['SR04']['Distance'])
var var1=sensors['SR04']['Distance']
var var2=var1+10
print (var1,var2)

var msg = string.format(",\"V1\":{\"V2\":%i,\"V3\":%5.3f,\"V4\":%5.4f,\"V5\":%5.3f}", 44, 33.9, 333.4, 55.6)
      tasmota.response_append(msg)
~~~

**Comment by s-hadinger** [comment]
Ha. Sorry for the confusion. When I mentioned GitHub, I was referring to Tasmota's repo, not Berry's.

https://github.com/arendst/Tasmota

**Comment by s-hadinger** [comment]
Now to answer your question, `tasmota.response_append()` only works in the method `json_append()` of a registered driver. This method is called at each teleperiod, this is when you get a chance to add your data.

I will add it in the documentation

**Comment by trlafleur** [comment]
so, what is the correct way to append or send a new MQTT ??

what I want to do, is get a sensor reading (or trigger one) make changes to
data, then send MQTT

thanks

~~ _/) ~~~~ _/) ~~~~ _/) ~~~~ _/) ~~

Tom Lafleur

**Comment by trlafleur** [comment]
will move to Tasmota...
my error...

~~ _/) ~~~~ _/) ~~~~ _/) ~~~~ _/) ~~

Tom Lafleur
